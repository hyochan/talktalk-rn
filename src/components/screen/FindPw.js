// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  ScrollView,
} from 'react-native';

import styled from 'styled-components/native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

const StyledContainer = styled.View`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const StyledScrollViewContainer = styled.View`
  justify-content: center;
  align-items: center;
`;

const StyledWrapper = styled.View`
  margin-top: 40;
  width: 78%;
`;

const StyledButtonWrapper = styled.View`
  width: 100%;
  align-items: flex-end;
  margin-top: 16;
`;

type Props = {
  navigation: any,
};
type State = {
  email: string,
  isLoading: boolean,
};

class Screen extends Component<Props, State> {
  static navigationOptions = {
    title: getString('FIND_PW'),
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      isLoading: false,
      email: '',
    };
  }

  render() {
    return (
      <StyledContainer>
        <StatusBar isDarkContent={false}/>
        <StyledScrollView
          contentContainerStyle={{
            alignItems: 'center',
            justifyContent: 'center',
          }}
        >
          <StyledWrapper>
            <TextInput
              style={{ marginTop: 8 * ratio }}
              txtLabel={getString('EMAIL')}
              txtHint={ getString('EMAIL') }
              txt={ this.state.email }
              onTextChanged={ (text) => this.onTextChanged('EMAIL', text)}
              isPassword={ true }
            />
            <StyledButtonWrapper>
              <Button
                id='send_link'
                isLoading={this.state.isLoading}
                onPress={() => this.onSendLink()}
              >{getString('SEND_LINK')}</Button>
            </StyledButtonWrapper>
          </StyledWrapper>
        </StyledScrollView>
      </StyledContainer>
    );
  }

  onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        this.setState({ email: text });
        break;
    }
  }

  goBack = () => {
    this.props.navigation.goBack();
  }

  onSendLink = () => {
    console.log('onSendLink');
    this.setState({
      isLoading: true,
    }, async() => {
      this.setState({ isLoading: false });
    });
  }
}

export default Screen;
