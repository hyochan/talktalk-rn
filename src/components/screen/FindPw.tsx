import React, { Component, useState, useEffect } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  ScrollView,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';

import styled from 'styled-components/native';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

const StyledContainer = styled.View`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const StyledScrollViewContainer = styled.View`
  justify-content: center;
  align-items: center;
`;

const StyledWrapper = styled.View`
  margin-top: 40;
  width: 78%;
`;

const StyledButtonWrapper = styled.View`
  width: 100%;
  align-items: flex-end;
  margin-top: 16;
`;

interface IProps {
  navigation: NavigationScreenProp<NavigationStateRoute<any>>;
}

interface IState {
  email: string;
  isLoading: boolean;
}

function Screen(props: IProps) {
  const [isLoading, setLoading] = useState(false);
  const [email, setEmail] = useState('');

  useEffect(() => {
    if (isLoading) {
      setLoading(false);
    }
  }, [isLoading]);

  const onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        setEmail(text);
        break;
    }
  };

  const onSendLink = () => {
    console.log('onSendLink');
    setLoading(true);
  };

  return (
    <StyledContainer>
      <StatusBar isDarkContent={false}/>
      <StyledScrollView
        contentContainerStyle={{
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <StyledWrapper>
          <TextInput
            testID='input_email'
            style={{ marginTop: 8 * ratio }}
            txtLabel={getString('EMAIL')}
            txtHint={ getString('EMAIL') }
            txt={ email }
            onTextChanged={ (text) => onTextChanged('EMAIL', text)}
            isPassword={ true }
          />
          <StyledButtonWrapper>
            <Button
              testID='btnSendLink'
              isLoading={isLoading}
              onPress={() => onSendLink()}
            >{getString('SEND_LINK')}</Button>
          </StyledButtonWrapper>
        </StyledWrapper>
      </StyledScrollView>
    </StyledContainer>
  );
}

export default Screen;
