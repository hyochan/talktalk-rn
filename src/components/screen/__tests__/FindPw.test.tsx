import 'react-native';
import * as React from 'react';
import FindPw from '../FindPw';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const props = {
  navigation: {
    navigate: jest.fn(),
  },
};
const component: React.ReactElement = <FindPw {...props}/>;

describe('[FindPw] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[FindPw] interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;
  let testingLib: any;

  beforeAll(() => {
    rendered = renderer.create(component);
    root = rendered.root;
    testingLib = render(component);
  });

  it('should invoke changeText event handler when email changed ', () => {
    const emailInput = testingLib.getByTestId('input_email');
    emailInput.props.onTextChanged('email test');
    expect(emailInput.props.txt).toEqual('email test');
  });

  it('should call [sendLink] on button click', () => {
    const btnSendLink = testingLib.getByTestId('btnSendLink');
    fireEvent(btnSendLink, 'press');
    expect(btnSendLink.props.isLoading).toEqual(false);
    btnSendLink.props.onPress();
    expect(btnSendLink.props.isLoading).toEqual(true);
  });
});
