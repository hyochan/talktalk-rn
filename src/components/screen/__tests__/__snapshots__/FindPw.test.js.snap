// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<Styled(View)>
  <Shared
    isDarkContent={false}
  />
  <Styled(ScrollView)
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Styled(View)>
      <Shared
        isPassword={true}
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 16.666666666666664,
          }
        }
        txt=""
        txtHint="Email"
        txtLabel="Email"
      />
      <Styled(View)>
        <Button
          activeOpacity={0.5}
          height={60}
          id="send_link"
          indicatorColor="white"
          isDisabled={false}
          isLoading={false}
          isWhite={false}
          onPress={[Function]}
          width={136}
        >
          Send Link
        </Button>
      </Styled(View)>
    </Styled(View)>
  </Styled(ScrollView)>
</Styled(View)>
`;
